pkg/apis/autoscaling/v1alpha1/pa_validation.go

```diff
@@ -48,7 +50,28 @@ func (rs *PodAutoscalerSpec) Validate(ctx context.Context) *apis.FieldError {
 	} else if err := servingv1alpha1.ValidateContainerConcurrency(rs.ContainerConcurrency, rs.ConcurrencyModel); err != nil {
 		errs = errs.Also(err)
 	}
-	return errs
+	return errs.Also(validateSKSFields(rs))
+}
+
+func validateSKSFields(rs *PodAutoscalerSpec) *apis.FieldError {
+	var all *apis.FieldError
+	switch rs.ProtocolType {
+	case servingv1alpha1.RevisionProtocolH2C, servingv1alpha1.RevisionProtocolHTTP1:
```

Comment:
Since this is it's own type, we should simply add a Validate method that checks enum values.  We can still special case empty here.

Reply:
I think it's overkill. But sure.

Reply:
My concern is a proliferation of `switch` statements with the enum values, and if we expand the space we'll invariably forget to update one :(

Reply:
This should be gone.

Reply:
I also think we should have just a single place. May be, we'll home it outside specific subsystem API package.
But that's for a different pr, I think.